grails-app/services/org/pih/warehouse/picklist/PicklistService.groovy

```diff
@@ -322,4 +324,32 @@ class PicklistService {
             order("lastUpdated", "desc")
         }
     }
+
+    void revertPick(String id, String itemId) {
+        Picklist picklist = Picklist.get(id)
+        RequisitionItem requisitionItem = RequisitionItem.get(itemId)
+        List<PicklistItem> picklistItemsToRemove = PicklistItem.findAllByRequisitionItem(requisitionItem)
+
+        if (!picklist) {
+            throw new ObjectNotFoundException(id, Picklist.class.toString())
+        }
```

Comment:
I would call this just after attempting to get a picklist, to avoid some unnecessary queries (e.g. the one from 331th), if they are not gonna be used either way.